be2-scala/src/main/scala/ch/epfl/pop/model/objects/Channel.scala

```diff
@@ -20,6 +20,15 @@ final case class Channel(channel: String) {
     case _ => None
   }
 
+  /**
+   * Extract the channel id from a Channel (i.e. the last part of the path)
+   *
+   * @return the id of the Channel
+   *
+   * @example extractChannelId(Channel("/root/mEKXWFCMwb") == Hash(Base64Data("mEKXWFCMwb"))
+   */
+  def extractChannelId: Hash = Hash(Base64Data(channel.split(Channel.SEPARATOR).last))
```

Comment:
the naming is a bit confusing.
technically you're only extracting last component, but for an election, the path will be `/root/<lao id>/<election id>` , so you'd only be getting the `election_id` here.
where is this used?

Reply:
In the case of an election, it's used to retrieve the id in order to compare it in the validation stage

Reply:
Renammed it